lib/beaker/host.rb

```diff
@@ -209,6 +209,20 @@ def graceful_restarts?
       graceful
     end
 
+    # Returns true if the host is running in FIPS mode.
+    #
+    # We currently only test FIPS mode on Redhat 7. Other detection
+    # modes should be added here if we expand FIPS support to other
+    # platforms.
+    def fips_mode?
+      case self['platform']
+      when /el-7/
+        execute("cat /proc/sys/crypto/fips_enabled") == "1"
```

Comment:
Does this file contain a trailing newline that may need to be chomped? Asking because the puppet version only [reads the first byte](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/platform.rb#L24-L26)

Reply:
I can do that to be safe. I'm testing this right now though, and this version does work.